lib/resty/upstream/healthcheck.lua

```diff
@@ -651,6 +651,78 @@ local function gen_peers_status_info(peers, bits, idx)
     return idx
 end
 
+local function gen_peers_prometheus_status_info(peers, bits, idx, u, role)
+    local npeers = #peers
+    for i = 1, npeers do
+        idx = idx + 1
+        local peer = peers[i]
+        if peer.down then
+            bits[idx] = string.format("nginx_upstream_status{name=\"%s\",endpoint=\"%s\", status=\"DOWN\", role=\"%s\"} %d \n", u, peer.name, role,os.time())
```

Comment:
Is the `os.time()` redundant? Since the prometheus data is time series data, which already contains the timestamp.
Maybe we can use hard coded `1`, change the metric type to `Gauge` and change the name to `nginx_upstream_status_info` (like this one: https://www.robustperception.io/exposing-the-software-version-to-prometheus).

Reply:
It`s a good idea!  It is more convenient to use gauge than counter.